scrapy/utils/asyncio.py

```diff
@@ -0,0 +1,23 @@
+def install_asyncio_reactor():
+    """ Tries to install AsyncioSelectorReactor
+    """
+    try:
+        import asyncio
+        from twisted.internet import asyncioreactor
+    except ImportError:
+        pass
```

Comment:
If asyncio is explicitly requested but not available, shouldn’t we raise an Exception (maybe let the `ImportError` raise) or at the very least log an error?

Or is this better done from the calling code instead?

Reply:
It can be discussed.

`install_asyncio_reactor` is called in various "early" places if ASYNCIO_ENABLED is on. If it fails, we can abort the spider or we can log the warning. Currently we log it, with ERROR, in `log_scrapy_info()` when the spider starts, which is not the place where it's installed but later.

Reply:
We also compare `is_asyncio_reactor_installed()` and `ASYNCIO_ENABLED` when coroutines are used, and raise an exception (which I think will break the spider).